yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,253 @@ def _real_extract(self, url):
+        },
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/Video?id=0dc69bda-079b-4070-a7db-a8da1a06a9c7',
+        'only_matching': True,
+    }, {
+        'url': 'https://medius.microsoft.com/Embed/video-nc/fe823a91-959c-465b-96d4-8f4db624f72c',
+        'only_matching': True,
+    }]
+
+    def _extract_subtitle(self, webpage, video_id):
+        captions = traverse_obj(
+            self._search_json(r'const\s+captionsConfiguration\s*=', webpage, 'captions', video_id, default=None),
+            ('languageList', ..., {
+                'url': ('src', {url_or_none}),
+                'tag': ('srclang', {str}),
+                'name': ('kind', {str}),
+            })) or traverse_obj(
+            re.findall(r'var\s+file\s+=\s+\{[^}]+\'(https://[^\']+\.vtt\?[^\']+)', webpage),
+            (lambda _, v: url_or_none(v), {lambda x: {'url': x, 'tag': x.split('.vtt?')[0].split('_')[-1]}}))
+
+        return self._sub_to_dict(captions)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(f'https://medius.microsoft.com/Embed/video-nc/{video_id}', video_id)
+
+        ism_url = self._search_regex(r'StreamUrl\s*=\s*"([^"]+manifest)"', webpage, 'ism url')
+
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+            'formats': self._extract_ism(ism_url, video_id),
+            'thumbnail': self._og_search_thumbnail(webpage),
+            'subtitles': self._extract_subtitle(webpage, video_id),
+        }
+
+
+class MicrosoftLearnPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://learn\.microsoft\.com/(?:[\w\-]+/)?(?P<type>shows|events)/(?P<id>[\w\-]+)/?(?:[?#]|$)'
+    _TESTS = [{
+        'url': 'https://learn.microsoft.com/en-us/shows/bash-for-beginners',
+        'info_dict': {
+            'id': 'bash-for-beginners',
+            'title': 'Bash for Beginners',
+            'description': 'md5:16a91c07222117d1e00912f0dbc02c2c',
+        },
+        'playlist_count': 20,
+    }, {
+        'url': 'https://learn.microsoft.com/en-us/events/build-2022',
+        'info_dict': {
+            'id': 'build-2022',
+            'title': 'Microsoft Build 2022 - Events',
+            'description': 'md5:c16b43848027df837b22c6fbac7648d3',
+        },
+        'playlist_count': 201,
+    }]
+
+    def _entries(self, url_base, video_id):
+        skip = 0
+        while True:
+            playlist_info = self._download_json(url_base, video_id, f'Downloading entries {skip}', query={
+                'locale': 'en-us',
+                '$skip': skip,
+            })
+            items = traverse_obj(playlist_info, (
+                'results', ..., 'url', {lambda x: self.url_result(f'https://learn.microsoft.com/en-us{x}')}))
+            yield from items
+            skip += len(items)
+            if skip >= playlist_info['count'] or not items:
```

Comment:
We can't construct the url results like this. In traversal, the transformation is always called, so the lambda f-string could produce URLs like `'https://learn.microsoft.com/en-usNone'` if one of the dicts in the list doesn't have an `url` field.

```suggestion
            url_paths = traverse_obj(playlist_info, ('results', ..., 'url', {str}))
            for url_path in url_paths:
                yield self.url_result(f'https://learn.microsoft.com/en-us{url_path}')
            skip += len(url_paths)
            if skip >= playlist_info.get('count', 0) or not url_paths:
```